### PR TITLE
Fix crosswalk warning too early 18102

### PR DESCRIPTION
**Pull request by Dima-1**

### CODE REVIEW

**Comment by ivanPyrohivskyi** [comment]
For test use Alarm.obf from resoirces

**Review by ivanPyrohivskyi — APPROVED** [review]

File: OsmAnd/src/net/osmand/plus/helpers/WaypointHelper.java [hunk 1]
```diff
@@ -245,28 +245,38 @@ public AlarmInfo getMostImportantAlarm(SpeedConstants sc, boolean showCameras) {
 				AlarmInfo inf = (AlarmInfo) lwp.point;
 				int currentRoute = route.getCurrentRoute();
 				if (inf.getLocationIndex() < currentRoute && inf.getLastLocationIndex() != -1
-						&& inf.getLastLocationIndex() < currentRoute) {
+						&& inf.getLastLocationIndex() < currentRoute
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/helpers/WaypointHelper.java:248]: not obvious order evaluations with && and ||

File: OsmAnd/src/net/osmand/plus/helpers/WaypointHelper.java [hunk 2]
```diff
@@ -245,28 +245,38 @@ public AlarmInfo getMostImportantAlarm(SpeedConstants sc, boolean showCameras) {
...
 					}
+					int distance = route.getDistanceFromPoint(currentRoute) - route.getDistanceFromPoint(inf.getLocationIndex() - 1);
+					Location nextLocation = route.getNextRouteLocation();
+					Location nextInfoLocation = new Location("", inf.getLatitude(), inf.getLongitude());
 					Location lastKnownLocation = app.getRoutingHelper().getLastProjection();
-					int d = (int) Math.max(0.0, MapUtils.getDistance(lastKnownLocation.getLatitude(), lastKnownLocation.getLongitude(),
-							inf.getLatitude(), inf.getLongitude()) - lwp.getDeviationDistance());
-					if (inf.getLocationIndex() == currentRoute && d > 10) {
-						return null;
+					if (lastKnownLocation != null) {
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/helpers/WaypointHelper.java:262]: - lwp.getDeviationDistance() is missing

Comment (vshcherb) [reply OsmAnd/src/net/osmand/plus/helpers/WaypointHelper.java:262]: - distance calculation is a bit complicated - clarification needed

**Pull request closed without merging**